src/classic/clvm/casts.rs

```diff
@@ -125,6 +125,7 @@ pub fn bigint_to_bytes(v_: &Number, option: Option<TConvertOption>) -> Result<By
     let div = if signed { bi_one() } else { bi_zero() };
     let b32: u64 = 1_u64 << 32;
     let bval = b32.to_bigint().unwrap();
+    let (_sign, u32_digits) = v.to_u32_digits();
```

Comment:
it seems to me that almost everything in this function could be replaced by a call to `v_.to_signed_bytes_be()` + a fix-up to prepend a leading 0 in case it's negative.

See this function: https://github.com/Chia-Network/clvm_rs/blob/main/src/number.rs#L8

Reply:
Thanks, I hadn't seen that part, but will simplify it.

Reply:
It passes all the tests i have, but it was used unsigned in one place; as_path().  I _think_ that encoding is fine but need to write more tests.

Reply:
added a lot more testing.

Reply:
it seems to me that you're making it a lot harder for yourself by converting the bigint to `u32_digits()` rather than big endian bytes directly. it makes me a bit nervous because of the increased complexity and the increased number of edge cases that needs to be tested.

You say this is used for serializing a node path, which are always unsigned and never needs a leading 0 byte to make them unsigned, right?

If so, I can think of a few simplifications that would both make this function smaller but primarily bring down the number of test cases you need. (1) remove all handling of negative numbers in here, and just `assert!()` that the number is positive. (2) You could use a `BigUint` to represent the number, but I don't think it really makes a difference when converting to bytes. I think in this case even using `BigInt::to_be_bytes()` doesn't add a leading 0 for numbers that otherwise would be negative. So, as far as I understand, that's exactly what you want here.

Reply:
Apologies, i should have done much more up front porting this.  I reintroduced it after failing a recompile check without.
I am constantly trying to balance being responsive to people, getting things improved and delivering stuff and have cut that choice wrong a lot of time.

I put this back (it was here before) mainly to reduce the size of what i'm doing in this patch and hopefully to avoid creating a lot more work, but probably trying to save in the wrong place.

Let me look at rewriting it, apologies.

Reply:
It's also the case that the rest of the code is organized around this, causing more churn.
I'll endeavor to separate out the uses.

Reply:
rereading the node path code, i need to make sense of it a bit more and rewrite a chunk of it as well.  was hoping to avoid making too many changes in here (and potentially make things worse).

Reply:
the option is actually important here to the way nodepath operates (at least in the ported form).  I see what the typescript code was meant to accomplish ... it normalizes the integer representation by first doing a signed conversion and then reading as unsigned.  the final conversions out when being written are unsigned.

Reply:
Ok, presuming this version passes all checks, i think that's what you're looking for.